app/controllers/katello/api/v2/products_controller.rb

```diff
@@ -28,25 +30,19 @@ class Api::V2::ProductsController < Api::V2::ApiController
     param :custom, :bool, :desc => N_("Filter products by custom")
     param_group :search, Api::V2::ApiController
     def index
-      options = {
-        :filters => [],
-        :load_records? => true
-      }
-      products = Product.readable.where(:organization_id => @organization.id)
-      products = products.where(:provider_id => @organization.anonymous_provider.id) if params[:custom]
-
-      ids = products.pluck(:id)
-      ids = filter_by_subscription(ids, params[:subscription_id]) if params[:subscription_id]
-      ids = filter_by_activation_key(ids, @activation_key) if @activation_key
-      ids = filter_by_system(ids, @system) if @system
-
-      options[:filters] << {:terms => {:id => ids}}
-      options[:filters] << {:term => {:name => params[:name]}} if params[:name]
-      options[:filters] << {:term => {:enabled => ::Foreman::Cast.to_bool(params[:enabled])}} if params[:enabled]
-      options.merge!(sort_params)
-      options[:includes] = [:sync_plan, :provider]
+      options = {:includes => [:sync_plan, :provider]}
+      respond(:collection => scoped_search(index_relation.uniq, :name, :desc, options))
+    end
 
-      respond(:collection => item_search(Product, params, options))
+    def index_relation
+      query = Product.readable.where(:organization_id => @organization.id)
+      query = query.where(:provider_id => @organization.anonymous_provider.id) if params[:custom]
+      query = query.where(:name => params[:name]) if params[:name]
+      query = query.enabled if params.key?(:enabled)
+      query = filter_by_activation_key(query, @activation_key) if @activation_key
```

Comment:
Is it worth keeping this function? This appears to be the only place that uses it. Same for the the three below this.